client/inputs_http_event_collector.go

```diff
@@ -60,7 +61,7 @@ func (client *Client) DeleteHttpEventCollectorObject(name, owner, app string) (*
 
 // services/data/inputs/http
 func (client *Client) ReadAllHttpEventCollectorObject() (*http.Response, error) {
-	endpoint := client.BuildSplunkURL(nil, "servicesNS", "-", "-", "data", "inputs", "http")
+	endpoint := client.BuildSplunkURL(nil, "servicesNS", "nobody", "search", "data", "inputs", "http")
```

Comment:
With HEC inputs being in the global context, they don't really need a namespace.  This could potentially be:

`endpoint := client.BuildSplunkURL(nil, "services", "data", "inputs", "http")`